src/wp-includes/option.php

```diff
@@ -827,7 +825,16 @@ function update_option( $option, $value, $autoload = null ) {
 	);
 
 	if ( null !== $autoload ) {
-		$update_args['autoload'] = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+		$update_args['autoload'] = determine_option_autoload_value( $option, $serialized_value, $autoload );
+	} else {
+		$raw_autoload = $wpdb->get_var( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name = %s LIMIT 1", $option ) );
+		$allow_values = array( 'on', 'off', 'auto-on', 'auto-off' );
```

Comment:
Only values that are not explicit should be overwritten. Also, `auto` needs to be included.
```suggestion
		$allow_values = array( 'auto-on', 'auto-off', 'auto' );
```